Xplat/src/main/java/vazkii/botania/mixin/EntityMixin.java

```diff
@@ -33,6 +36,10 @@ private void checkInvulnerabilities(DamageSource source, CallbackInfoReturnable<
 			} else if (CrimsonPendantItem.onDamage(self, source)) {
 				cir.setReturnValue(true);
 			}
+		} else {
+			if (source.is(BotaniaTags.DamageTypes.NONLIVING_IMMUNE)) {
+				cir.setReturnValue(true);
+			}
 		}
```

Comment:
As mentioned on chat, I don't think a NONLIVING_IMMUNE tag is the right solution.

Let's invert this and scope down:
```suggestion
		}
		if (source.is(BotaniaDamageTypes.KEY_EXPLOSION) && ((Object) this) instanceof ItemEntity) { // or if you'd like, a botania:key_explosion_immune entity tag, like vanilla's minecraft:fall_damage_immune entity tag
		  cir.setReturnValue(true);
		}
```

Reply:
Switched to two tags, one for IS_KEY (similar to vanilla's IS_FALL) and one for key_immune (similar to fall_damage_immune)